core/trino-main/src/main/java/io/trino/operator/scalar/timetz/TimeZone.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.scalar.timetz;
+
+import io.airlift.slice.Slice;
+import io.trino.spi.function.Description;
+import io.trino.spi.function.LiteralParameters;
+import io.trino.spi.function.ScalarFunction;
+import io.trino.spi.function.SqlType;
+import io.trino.spi.type.LongTimeWithTimeZone;
+import io.trino.spi.type.StandardTypes;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.spi.type.DateTimeEncoding.unpackOffsetMinutes;
+import static io.trino.type.DateTimes.MINUTES_PER_HOUR;
+
+@Description("Time zone of the given time")
+@ScalarFunction("timezone")
+public final class TimeZone
+{
+    private TimeZone() {}
+
+    /**
+     * Extracts the time zone from a `time(p) with time zone` as a VARCHAR.
+     * This function takes a packed long value representing
+     * the time with time zone and extracts the associated time zone offset.
+     * <p>
+     *
+     * @param timeWithTimeZone the packed long
+     * representing a `time(p) with time zone`
+     * @return the time zone offset in the format
+     * "+HH:mm" or "-HH:mm" as a VARCHAR
+     */
+    @LiteralParameters("p")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice extractTimeZoneFromTime(
+            @SqlType("time(p) with time zone") final long timeWithTimeZone)
+    {
+        long totalOffsetMinutes = unpackOffsetMinutes(timeWithTimeZone);
+
+        long hours = Math.abs(totalOffsetMinutes) / MINUTES_PER_HOUR;
+        long minutes = Math.abs(totalOffsetMinutes) % MINUTES_PER_HOUR;
+
+        String sign = totalOffsetMinutes < 0 ? "-" : "+";
+
+        return utf8Slice(String.format("%s%02d:%02d", sign, hours, minutes));
```

Comment:
I'd prefer to avoid `String.format` as it's rather slow and produces a lot of garbage and go with something like this instead (and in the other function as well):

```
        char sign = timeWithTimeZone.getOffsetMinutes() > 0 ? '+' : '-';
        long totalOffsetMinutes = abs(timeWithTimeZone.getOffsetMinutes());

        String hours = Long.toString(LongMath.divide(totalOffsetMinutes, MINUTES_PER_HOUR, FLOOR));
        String minutes = Long.toString(LongMath.mod(totalOffsetMinutes), MINUTES_PER_HOUR);

        return utf8Slice(sign + padStart(hours, 2, '0') + ':' +
                padStart(minutes, 2, '0'));
```